lib/ddtrace/transport/http/transport.rb

```diff
@@ -0,0 +1,92 @@
+module Datadog
```

Comment:
Ready for review again; the introduction of this class is the main change.
It adds a layers between the `writer` and `client` that handles chunking and API downgrading.